main.go

```diff
@@ -88,20 +84,15 @@ func handleUserRequest(
 			panic(err)
 		}
 
-		userId, err := databaseutil.CreateUser(
+		_, err := userservice.CreateUser(
```

Comment:
this line i think highlights that create user is improperly named. It should be called add user to database, or it should create the user as an object return it then have a second line that adds it to the database. Here esp with the anonymous var it looks like we are specifically throwing away the newly created user.

Reply:
hmm... I think we have very different mental models for the code. In my mind, a user is something that only truly exists in the database. So saying "create" user to me means that it is being inserted into the database. I would think it very misleading if `createUser` just constructed an object in the code.

I'm also just very used to the idea that `createX` inserts it into the database and returns the id. Would it be clearer if it was `recordX` or something?

Another thing that may be causing confusion here is that we're conflating two things in the user table. We could theoretically have a user table that just has displayName (and perhaps emailAddress), and then an authentication or password table with userId and password

Reply:
Hmmm, ya i think i'm used to `create` in the more of an OOP way. So like `createX` would be a static method on a class that would return a new object of that class. Also to exacerbate when dealing with db i'm used to transactions so you'd have to call save at the end of your operations for the information to be stored into the db. I highly prefer `recordX` over `createX`.  What are your feelings on `saveX` or `storeX`.

Reply:
Change to `storeX` and no longer return id

Reply:
`storeNewX`